src/neo/SmartContract/Native/Tokens/NeoToken.cs

```diff
@@ -224,7 +268,14 @@ private bool Vote(ApplicationEngine engine, UInt160 account, ECPoint voteTo)
                 CandidateState state_validator = storage_validator.GetInteroperable<CandidateState>();
                 state_validator.Votes -= state_account.Balance;
                 if (!state_validator.Registered && state_validator.Votes.IsZero)
+                {
+                    DistributeGas(engine, account, state_account);
+                    UInt160 voteeAddr = Contract.CreateSignatureContract(state_account.VoteTo).ScriptHash;
+                    foreach (var (rewardKey, _) in engine.Snapshot.Storages.Find(CreateStorageKey(Prefix_VoterRewardPerCommittee).Add(voteeAddr).ToArray()))
+                        engine.Snapshot.Storages.Delete(rewardKey);
+
                     engine.Snapshot.Storages.Delete(key);
+                }
             }
             state_account.VoteTo = voteTo;
```

Comment:
We should call `DistributeGas()` before setting `state_account.VoteTo`.

Reply:
Done